vioscsi/vioscsi.c

```diff
@@ -1011,147 +1006,22 @@ VioScsiInterrupt(
-        }
-        else
-        {
-            ProcessBuffer(DeviceExtension, QUEUE_TO_MESSAGE(VIRTIO_SCSI_REQUEST_QUEUE_0), InterruptLock);
-        }
-    }
-
-    RhelDbgPrint(TRACE_LEVEL_VERBOSE, " isInterruptServiced = %d\n", isInterruptServiced);
-    return isInterruptServiced;
-}
-
-static BOOLEAN
-VioScsiMSInterruptWorker(
-    IN PVOID  DeviceExtension,
-    IN ULONG  MessageID
-    )
-{
-    PVirtIOSCSICmd      cmd;
-    PVirtIOSCSIEventNode evtNode;
-    unsigned int        len;
-    PADAPTER_EXTENSION  adaptExt;
-    PSRB_TYPE           Srb = NULL;
-    ULONG               intReason = 0;
-
-    adaptExt = (PADAPTER_EXTENSION)DeviceExtension;
-
-    RhelDbgPrint(TRACE_LEVEL_VERBOSE,
-                 " MessageID 0x%x\n", MessageID);
-
-    if (MessageID >= QUEUE_TO_MESSAGE(VIRTIO_SCSI_REQUEST_QUEUE_0))
-    {
-        DispatchQueue(DeviceExtension, MessageID);
-        return TRUE;
-    }
-    if (MessageID == 0)
-    {
-       return TRUE;
-    }
-    if (MessageID == QUEUE_TO_MESSAGE(VIRTIO_SCSI_CONTROL_QUEUE))
-    {
-        if (adaptExt->tmf_infly)
-        {
-           while((cmd = (PVirtIOSCSICmd)virtqueue_get_buf(adaptExt->vq[VIRTIO_SCSI_CONTROL_QUEUE], &len)) != NULL)
-           {
-              VirtIOSCSICtrlTMFResp *resp;
-              Srb = (PSRB_TYPE)(cmd->srb);
-              ASSERT(Srb == (PSRB_TYPE)&adaptExt->tmf_cmd.Srb);
-              resp = &cmd->resp.tmf;
-              switch(resp->response) {
-              case VIRTIO_SCSI_S_OK:
-              case VIRTIO_SCSI_S_FUNCTION_SUCCEEDED:
-                 break;
-              default:
-                 RhelDbgPrint(TRACE_LEVEL_ERROR, " Unknown response %d\n", resp->response);
-                 ASSERT(0);
-                 break;
-              }
-              StorPortResume(DeviceExtension);
-           }
-           adaptExt->tmf_infly = FALSE;
-        }
-        return TRUE;
-    }
-    if (MessageID == QUEUE_TO_MESSAGE(VIRTIO_SCSI_EVENTS_QUEUE)) {
-        while((evtNode = (PVirtIOSCSIEventNode)virtqueue_get_buf(adaptExt->vq[VIRTIO_SCSI_EVENTS_QUEUE], &len)) != NULL) {
-           PVirtIOSCSIEvent evt = &evtNode->event;
-           switch (evt->event) {
-           case VIRTIO_SCSI_T_NO_EVENT:
-              break;
-           case VIRTIO_SCSI_T_TRANSPORT_RESET:
-              TransportReset(DeviceExtension, evt);
-              break;
-           case VIRTIO_SCSI_T_PARAM_CHANGE:
-              ParamChange(DeviceExtension, evt);
-              break;
-           default:
-              RhelDbgPrint(TRACE_LEVEL_ERROR, " Unsupport virtio scsi event %x\n", evt->event);
-              break;
-           }
-           SynchronizedKickEventRoutine(DeviceExtension, evtNode);
-        }
-        return TRUE;
+    if ((virtio_read_isr_status(&adaptExt->vdev) == 1) || adaptExt->dump_mode) {
+        return ProcessQueue(DeviceExtension, VIRTIO_SCSI_REQUEST_QUEUE_0 + VIRTIO_SCSI_MSI_CONTROL_Q_OFFSET, "ProcessQueue");
```

Comment:
personal nit: can we wrap such long lines, like:
```
return blah(long_name_arg1,
            long_name_arg2,
            long_name_arg3)
```

Reply:
@JonKohler @kostyanf14 
Clang formater is coming soon (already added for Balloon and FWCfg drivers).

The long list of parameters will be wrapped

Reply:
@YanVugenfirer @JonKohler @kostyanf14 

> Clang formatter is coming soon ...

Will it also remove the indent on the logger macros, e.g. `#if !defined(RUN_UNCHECKED)`...?
It would be good if it didn't.
Same for line / ASCII art as mentioned elsewhere...

It would seem I'm used to the wider screen...
...but can understand it helps to shrink the cols when using GitHub to review.   ;^D

Reply:
@benyamin-codez 

>Will it also remove the indent on the logger macros, e.g. #if !defined(RUN_UNCHECKED)...?

Yes, and this is as MS does in driver examples. 

>Same for line / ASCII art as mentioned elsewhere...

clang-format will be disabled for these specific lines.

Reply:
@kostyanf14 @JonKohler 

> > Will it also remove the indent on the logger macros, e.g. `#if !defined(RUN_UNCHECKED)`...?
>
> Yes, and this is as MS does in driver examples.

Technically correct, but so aesthetically displeasing... 8^(

Reply:
I think I'll leave this for the clang formatter...